src/sagas/pushNotification.js

```diff
@@ -246,10 +177,21 @@ export function* onAppInitialization() {
     return;
   }
 
-  yield call(installForegroundListener);
+  const deviceId = yield call(getDeviceId);
+  // const deviceId = yield call(getDeviceId);
+  if (!deviceId) {
+    console.debug('Halting push notification initialization because the device id is null.');
+    yield put(onExceptionCaptured(new Error('Device id is null')));
```

Comment:
Why don't you add this exception handler in the `getDeviceId` method? So if we use the method in another place in the future, we don't need to replicate the exception handler